consensus/src/dag/dag_driver.rs

```diff
@@ -153,30 +154,36 @@ impl DagDriver {
             (
                 highest_strong_links_round,
                 // unwrap is for round 0
-                dag_reader.get_strong_links_for_round(
-                    highest_strong_links_round,
-                    &self.epoch_state.verifier,
-                )
-                .unwrap_or_default(),
+                dag_reader
+                    .get_strong_links_for_round(
+                        highest_strong_links_round,
+                        &self.epoch_state.verifier,
+                    )
+                    .unwrap_or_default(),
             )
         };
 
         let minimum_delay = self
             .health_backoff
             .backoff_duration(highest_strong_link_round + 1);
-        self.round_state
+        let mut round_state_guard = self.round_state.lock().await;
+        round_state_guard
             .check_for_new_round(highest_strong_link_round, strong_links, minimum_delay)
             .await;
         Ok(())
     }
 
-    pub async fn enter_new_round(&mut self, new_round: Round) {
-        if self.round_state.current_round() >= new_round {
-            return;
+    pub async fn enter_new_round(&self, new_round: Round) {
```

Comment:
hmm, I find it's probably simpler to make this still mut self, having concurrent enter new round doesn't make much sense to me and may have weird implications

Reply:
We have an `Arc<DagDriver>` now, so this cannot be mut self.